internal/service/cloudprovideraccess/resource_cloud_provider_access_setup_test.go

```diff
@@ -136,20 +136,13 @@ func checkExists(resourceName string) resource.TestCheckFunc {
 		ids := conversion.DecodeStateID(rs.Primary.ID)
 		providerName := ids["provider_name"]
 		id := ids["id"]
-		roles, _, err := acc.Conn().CloudProviderAccess.ListRoles(context.Background(), ids["project_id"])
+		roles, _, err := acc.ConnV2().CloudProviderAccessApi.ListCloudProviderAccessRoles(context.Background(), ids["project_id"]).Execute()
 		if err != nil {
 			return fmt.Errorf(cloudprovideraccess.ErrorCloudProviderGetRead, err)
 		}
 		if providerName == "AWS" {
-			for i := range roles.AWSIAMRoles {
-				if roles.AWSIAMRoles[i].RoleID == id && roles.AWSIAMRoles[i].ProviderName == providerName {
-					return nil
-				}
-			}
-		}
-		if providerName == "AZURE" {
```

Comment:
The new SDK does not support returning AZURE data in the list API. Opening a Jira issue.

Reply:
You can use:
`role, _, err := acc.ConnV2().CloudProviderAccessApi.GetCloudProviderAccessRole(context.Background(), projectID, roleID).Execute()`

Reply:
can you clarify? this listRoles returns all the roles. How can I use the single GET API to get all the roles?

Reply:
This function checks if the AccessRole exists. 
In the current implementation, it uses the ListRoles method.
However, since we know the `project_id` and the `roleId` (`id := ids["id"]`) we can use those directly to get the access role with the GetCloudProviderAccessRole method and return an error if we get a 404.
E.g.: 

```golang
func checkExists(resourceName string) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		rs, ok := s.RootModule().Resources[resourceName]
		if !ok {
			return fmt.Errorf("not found: %s", resourceName)
		}
		if rs.Primary.Attributes["project_id"] == "" {
			return fmt.Errorf("no ID is set")
		}
		ids := conversion.DecodeStateID(rs.Primary.ID)
		providerName := ids["provider_name"]
		roleID := ids["id"]
		projectID := ids["project_id"]
		role, _, err := acc.ConnV2().CloudProviderAccessApi.GetCloudProviderAccessRole(context.Background(), projectID, roleID).Execute()
		if err != nil || role == nil {
			return fmt.Errorf("failed to find role: %s for cloud %s, in project %s, error: %v", roleID, providerName, projectID, err)
		}
		if role.GetProviderName() != providerName {
			return fmt.Errorf("provider name missmatch, expected %s, got %s", providerName, role.GetProviderName())
		}
		return nil
	}
}
```

Reply:
right, good point!